.github/workflows/release-new-release-published.yml

```diff
@@ -28,3 +29,59 @@ jobs:
           slack-optional-unfurl_links: false
           slack-text: |
             :woo-bounce: *<${{ github.event.release.html_url }}|WooCommerce ${{ github.event.release.name }}>* has been released! :rocket:
+
+  update-global-changelog:
+    name: 'Update changelog.txt after any stable release'
+    runs-on: ubuntu-latest
+    if: ${{ ! contains( github.event.release.tag_name, 'wc-beta-tester' ) && ! contains( github.event.release.tag_name, '-rc' )  }}
+
+    steps:
+      - name: 'Fetch release readme.txt'
+        env:
+          RELEASE_TAG: ${{ github.event.release.tag_name }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GH_REPO: ${{ github.repository }}
+        run: |
+          gh release download "$RELEASE_TAG" --pattern 'woocommerce.zip'
+          unzip woocommerce.zip
+
+          if [[ ! -e woocommerce/readme.txt  ]]; then
+            echo "::error::Could not find readme.txt in asset for release '$RELEASE_TAG'."
+            exit 1
+          fi
+      - name: 'Fetch latest changelog.txt and changelog script'
+        uses: 'actions/checkout@v4'
+        with:
+          ref: trunk
+          sparse-checkout: |
+            /changelog.txt
+            /.github/workflows/scripts/release-readme-to-changelog.php
+          sparse-checkout-cone-mode: false
+          path: trunk
+      - name: 'Update changelog.txt'
+        run: |
+          php ./trunk/.github/workflows/scripts/release-readme-to-changelog.php ./woocommerce/readme.txt ./trunk/changelog.txt
+      - name: 'Push changes and open PR'
+        env:
+          GH_TOKEN: ${{ github.token }}
+          GH_REPO: ${{ github.repository }}
+        run: |
+          # Set up Git.
+          git config --global user.name 'WooCommerce Bot'
```

Comment:
A bit more verbose than necessary: `git commit --author="woocommercebot <woocommercebot@users.noreply.github.com>"`.

PS: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#about-commit-email-addresses

Reply:
Gotcha, lets keep as it is then @jorgeatorres.